temporal-sdk/src/main/java/io/temporal/workflow/Workflow.java

```diff
@@ -1181,6 +1182,18 @@ public static Logger getLogger(String name) {
     return WorkflowInternal.getLastCompletionResult(resultClass, resultClass);
   }
 
+  /**
+   * Extract the latest failure from some previous run for this cron workflow. This is used in
+   * combination with cron schedule. A workflow can be started with an optional cron schedule. If
+   * any previous workflow has failed, this function returns that failure. If no previous workflows
+   * have failed, an empty optional is returned.
+   *
+   * @return The last {@link Failure} that occurred in this workflow, if there has been one.
+   */
+  public static Optional<Failure> getLastFailure() {
```

Comment:
Change the result type to Optional<RuntimeException> as neither Failure nor Payload are exposed to the workflow code. Use [failureToException](https://github.com/temporalio/sdk-java/blob/4bb23c3a660de6a2630901d0dc6c049d9742ebe1/temporal-sdk/src/main/java/io/temporal/failure/FailureConverter.java#L84) method to convert.

Reply:
Changed. Should we change the method name as well? Pertains to my question here:https://github.com/temporalio/sdk-go/pull/293#discussion_r528050388